### ISSUE TITLE
Unable to use `realm` property on List instances after v10.20.1

### ISSUE THREAD

**Issue by tfe** [realm/realm-swift#8156]
### How frequently does the bug occur?

Always

### Description

Hello, prior to v10.20.1 I was able to access the Realm instance that a List was managed by, similar to how you can still for Object, via an accessor like `pubic var realm: Realm?`.

It looks like the property that allowed this was removed in [this commit](https://github.com/realm/realm-swift/commit/868f1925ed38c24e0d2e79623785948e174971ab#diff-0165363078a304086de8efc5042f6a26f299760d001df8e433ee0db147c40c1eL42). Was that an intentional change? It looks like architecturally the Realm instance is still accessible from inside of List via its `rlmCollection`, it's just marked `internal` and not accessible.

If this was unintentional, it would be really helpful to restore that `realm` accessor. Its removal is making it difficult to upgrade from an older Realm version, since we'd have to find a workaround for accessing a List's Realm in the places we currently do that. As an example, when I have a List instance, this makes it easy to open a write transaction to append to the List without having to pass in or find the Realm instance separately.

Here's an example implementation based on the original version with updated property names:

```swift
/// The Realm which manages the list, or `nil` if the list is unmanaged.
public var realm: Realm? {
    return collection.realm.map(Realm.init)
}
```

Thanks for considering!

### Can you reproduce the bug?

Always

### Version

10.20.1

### What Atlas Services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

iOS

**Comment by tgoyne** [comment]
The implementation of this property was just moved to the `RealmCollection` protocol which `List` conforms to, and it still exists on `List`.

**Comment by tfe** [comment]
Hi @tgoyne, thanks for the quick reply! This is what I'm doing that made me think it had been removed (I see now that if I call `realm` directly, it works). I know this is out of scope for you at this point, but if you have any insight I'd be grateful... do you know why this would have stopped working after that version? Note that it doesn't complain about doing this for Object.

<img width="668" alt="Screen Shot 2023-03-01 at 10 34 43 AM" src="https://user-images.githubusercontent.com/21501/222233426-1f0f2418-b782-421c-a67e-4541235a5247.png">

It seems like that should work because (as you pointed out) List does still have a `realm` property.

**Comment by dianaafanador3** [comment]
Are you doing this within a SwiftUI View, it may be a conflict with SwiftUI `List` type and ours.
Have you tried
`extension RealmSwift.List: HasRealm {}`

**Comment by tfe** [comment]
@dianaafanador3 It's not, but I gave that a try anyway (because I have seen the issue you described elsewhere). Unfortunately it didn't help.

Thanks to you both for the help! Please let me know if you think of anything. For now I'll look for a workaround.

**Comment by dianaafanador3** [comment]
@tfe any luck with this, did you solve your issue?

**Comment by tfe** [comment]
@dianaafanador3 No, thanks for checking in. I had to move on so I just duplicated the functions I wanted to share between List and Object into separate extensions.

Did you try to reproduce? I was able to reproduce by creating a new Xcode project, adding RealmSwift via SPM, and pasting the following into a file:

```swift
import Foundation
import RealmSwift

protocol HasRealm {
    var realm: Realm? { get }
}

extension Object: HasRealm {}
extension List: HasRealm {} // Same error on this line

extension HasRealm {
    func doSomethingWithRealm() throws {
        guard let realm else { return }
        realm.beginWrite()
        // etc
        try realm.commitWrite()
    }
}
```

**Comment by leemaguire** [comment]
@tfe I can reproduce your issue. I will discuss this with the team.
